### PR TITLE
DEPR: enforce indexing deprecations

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -2661,27 +2624,22 @@ def need_slice(obj: slice) -> bool:
     )
 
 
-def check_deprecated_indexers(key) -> None:
+def check_dict_or_set_indexers(key) -> None:
     """Checks if the key is a deprecated indexer."""
```
Comment (mroeschke) [inline pandas/core/indexing.py:2628]: Could you update this docstring?

File: pandas/tests/frame/indexing/test_indexing.py [hunk 2]
```diff
@@ -1711,14 +1711,14 @@ def test_loc_on_multiindex_one_level(self):
...
 
 
-class TestDepreactedIndexers:
+class TestDeprecatedIndexers:
     @pytest.mark.parametrize(
         "key", [{1}, {1: 1}, ({1}, "a"), ({1: 1}, "a"), (1, {"a"}), (1, {"a": "a"})]
     )
     def test_getitem_dict_and_set_deprecated(self, key):
-        # GH#42825
+        # GH#42825 enforced in 2.0 enforced in 2.0
```
Comment (mroeschke) [inline pandas/tests/frame/indexing/test_indexing.py:1719]:
```suggestion
        # GH#42825 enforced in 2.0
```

File: pandas/core/indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexing.py]: updated + green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**